dotCMS/src/main/java/com/dotcms/rest/api/v1/index/ESIndexResource.java

```diff
@@ -67,392 +67,351 @@
+
+        System.setProperty("es.index.number_of_shards", String.valueOf(shards));
+        Logger.info(this, "Running Contentlet Reindex");
+
+        if(!DOTALL.equals(contentType)) {
+            ContentType type = APILocator.getContentTypeAPI(APILocator.systemUser()).find(contentType);
+            Logger.info(this.getClass(), "Starting reindex of " + type.name());
+            APILocator.getContentletIndexAPI().removeContentFromIndexByStructureInode(type.id());
+            APILocator.getContentletAPI().refresh(type);
+        }
+        else {
+            APILocator.getContentletAPI().refreshAllContent();
+        }
 
-    @PUT
-    @Path("/deactivate/{params:.*}")
-    public Response deactivateIndex(@Context HttpServletRequest httpServletRequest, @Context final HttpServletResponse httpServletResponse, @PathParam("params") String params) {
-        try {
-            InitDataObject init=auth(params,httpServletRequest,httpServletResponse);
-            String indexName = this.indexHelper.getIndexNameOrAlias(init.getParamsMap(),"index","alias",this.indexAPI);
+        return getReindexationProgress(request, response);
 
-            deactivateIndex(indexName);
 
-            return Response.ok().build();
-        } catch (Exception e) {
-            Logger.error(this.getClass(),"Exception trying to deactivate index: " + e.getMessage(), e);
-            return ResponseUtil.mapExceptionResponse(e);
+    }
+    
+    @CloseDBIfOpened
+    @DELETE
+    @JSONP
+    @NoCache
+    @Path("/reindex")
+    @Produces({MediaType.APPLICATION_JSON})
+    public Response stopReindexation(@Context final HttpServletRequest request, 
+                    @Context final HttpServletResponse response, 
+                    @DefaultValue("true") @QueryParam("switch") boolean switchMe)
+                    throws DotDataException {
+        final InitDataObject init = auth(request, response);
+        
+
+        if(switchMe) {
+            APILocator.getContentletIndexAPI().stopFullReindexationAndSwitchover();
+        }else {
+            APILocator.getContentletIndexAPI().stopFullReindexation();
         }
+        sendAdminMessage("reindex stopped", MessageSeverity.INFO,init.getUser(), 5000);
+        return getReindexationProgress(request, response);
     }
+    
+    
+    @CloseDBIfOpened
+    @DELETE
+    @JSONP
+    @NoCache
+    @Path("/{indexName: .*}")
+    @Produces({MediaType.APPLICATION_JSON})
+    public Response deleteIndex(@Context final HttpServletRequest request, @Context final HttpServletResponse response,
+                    @PathParam("indexName") final String indexName) throws DotDataException {
+        
+        final InitDataObject init = auth(request, response);
+
+        if(indexExists(indexName) ){
+            return Response.status(404).build();
+        }
+        
+        idxApi.delete(indexName);
 
+        String message = "Index:" + indexName + " deleted";
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'message' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5632525)